Add apps quickstart (#598)
saleor/saleor-docs: docs/developer/extending/apps/quickstart/getting-started.mdx

---
title: Getting started
---

## Overview

In this tutorial, we will walk you through the basic steps of building a Saleor app.

By the end of it, you will have built an **Abandoned Checkouts** app. It will fetch and display data about checkouts that for some reason haven't been turned into orders.

:::tip
Are you unsure about what a Saleor app is? You can find more information about them [here](developer/extending/apps/key-concepts.mdx).
:::

While building the app, we will cover:

- the basics of Saleor CLI
- creating an app and modifying its manifest
- installing and displaying the app in the Saleor Dashboard
- querying the data from the Saleor GraphQL API and using it to display a UI

## Prerequisites

### Knowledge

In the tutorial, we will introduce you to some Saleor app concepts. To make the most of them, you need to be familiar with several technologies.

The articles featured in this series will include the basics of:

- [TypeScript](https://www.typescriptlang.org/)
- [GraphQL](https://graphql.org/)
- [Next.js](https://nextjs.org/)

> Regardless of the purpose of this tutorial, we highly recommend learning them. These technologies make our standard stack and we use them in all sorts of different applications.

Author: ```suggestion
Setting aside the purpose of this tutorial, we highly recommend learning them. These technologies make up our standard stack and we use them in all sorts of different applications.
```

### Saleor CLI

> This tutorial relies on a technology we are deeply proud of: [Saleor CLI](cli.mdx). We will use it to spawn an app boilerplate.

Reviewer: cli.mdx is probably broken link

Reviewer: Ok, I was looking for it relative folder, now I see its in the root

To download and install Saleor CLI, run the following command:

```shell
npm i -g @saleor/cli
```

> After installation, you can display all the available commands by running the following in your terminal:

Author: ```suggestion
After the installation, you can display all available commands by running the following in your terminal:
```

```shell
saleor
```

### Saleor account

#### Login to your existing Saleor account

Before starting the development, please make sure you are logged in to your Saleor Cloud account. To log in let's type:

```shell
saleor login
```

The command will redirect you to a webpage with authentication. There, you need to pass in your login credentials. After a successful login, you can come back to the CLI.

:::tip

If your user is assigned to an organization and you have an existing project you want to build an app for, you can go all the way to the next page. See you there 👋!

:::

#### Create a new Saleor account

If you have no prior experience with Saleor, that's okay! Please create a new account with the command:

```shell
saleor register
```

The wizard will walk you through the registration process.

#### Create your organization

If you haven't done it already, we need to ensure that you are assigned to an organization before we continue. If that's not the case, please create an organization in the [Saleor Cloud dashboard](https://cloud.saleor.io/) after you have logged in.

#### Create your project

> Unless you want to assign your app to an existing project, we will need to create one.

Author: ```suggestion
Unless you want to assign your app to an existing project, we will need to create a new one.
```

> You can do so from the Saleor Cloud dashboard but since we made all the effort to install the Saleor CLI, let's make the most out of it!
> 

Author: ```suggestion
You can do so from the Saleor Cloud dashboard. But since we made all the effort to install the Saleor CLI, let's make the most of it!

```
